frontend/src/metabase/visualizations/shared/settings/cartesian-chart.ts

```diff
@@ -174,6 +175,23 @@ export const getDefaultXAxisScale = (
   return "ordinal";
 };
 
+const WATERFALL_UNSUPPORTED_X_AXIS_SCALES = ["pow", "log"];
+export const isXAxisScaleValid = (
+  series: RawSeries,
+  settings: ComputedVisualizationSettings,
+) => {
+  const isWaterfall = series[0].card.display === "waterfall";
+  const xAxisScale = settings["graph.x_axis.scale"];
+  if (
```

Comment:
We can just return the boolean directly, e.g.

`return !(isWaterfall && xAxisScale != null && WATERFALL_UNSUPPORTED_X_AXIS_SCALES.includes(xAxisScale)`

or

`return !isWaterfall || xAxisScale == null || !WATERFALL_UNSUPPORTED_X_AXIS_SCALES.includes(xAxisScale)`